protocol/communities/community_changes.go

```diff
@@ -79,7 +79,35 @@ func (c *CommunityChanges) HasMemberLeft(identity string) bool {
 	return ok
 }
 
-func EvaluateCommunityChanges(origin, modified *protobuf.CommunityDescription) *CommunityChanges {
+func EvaluateCommunityChanges(origin, modified *Community) *CommunityChanges {
+	changes := evaluateCommunityChangesByDescription(origin.Description(), modified.Description())
+
+	originTokenPermissions := origin.tokenPermissions()
+	modifiedTokenPermissions := modified.tokenPermissions()
+
+	// Check for modified or removed token permissions
+	for id, originPermission := range originTokenPermissions {
+		if modifiedPermission := modifiedTokenPermissions[id]; modifiedPermission != nil {
+			if !modifiedPermission.Equals(originPermission) {
+				changes.TokenPermissionsModified[id] = modifiedPermission
+			}
+		} else {
+			changes.TokenPermissionsRemoved[id] = originPermission
+		}
+	}
+
+	// Check for added token permissions
+	for id, permission := range modifiedTokenPermissions {
+		if _, ok := originTokenPermissions[id]; !ok {
+			changes.TokenPermissionsAdded[id] = permission
+		}
+	}
```

Comment:
why did you remove it from the evaluateCommunityChangesByDescription if this data from CommunityDescription?

Reply:
Because permissions are now obtained not only from the community description but also from events. See: https://github.com/status-im/status-go/blob/c3cac93229f25446c6203d4dea40b5d2eae960e9/protocol/communities/community.go#L1319